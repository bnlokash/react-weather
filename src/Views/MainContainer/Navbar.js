import React, { Component } from 'react';

import SELECTED from './navbarConstants.js';

import NavbarButton from './NavbarButton.js';

class Navbar extends Component {
  constructor(props){
    super();
    this.state = {
      selected: SELECTED.CURRENTLY
    };
    this.onClick = this.onClick.bind(this);

  }
  render() {
    return(
      <div className="w-100 dt dt--fixed">
        <NavbarButton id={SELECTED.CURRENTLY} text="Currently" selected={this.state.selected} onClick={this.onClick} />        
        <NavbarButton id={SELECTED.HOURLY} text="Hourly" selected={this.state.selected} onClick={this.onClick} />        
        <NavbarButton id={SELECTED.DAILY} text="Daily" selected={this.state.selected} onClick={this.onClick} />        </div>
    );
  }

  onClick(id) {
    this.setState({
      selected: id
    });
    this.props.selectFunc(id);
  }
}

export default Navbar;