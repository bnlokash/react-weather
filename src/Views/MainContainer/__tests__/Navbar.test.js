import React from 'react';
import renderer from 'react-test-renderer';

import Navbar from '../Navbar';
import SELECTED from '../navbarConstants';

test('Button text changes class when clicked', ()=>{
  let testState = (() =>{
    let _selected = null;
    return {
      setState: function(selected) {
        _selected = selected;
      }
    }
  })();

  const component = renderer.create(
    <Navbar selectFunc={testState.setState}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[2].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot(); 

});
