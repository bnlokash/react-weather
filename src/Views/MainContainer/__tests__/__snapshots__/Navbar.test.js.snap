// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button text changes class when clicked 1`] = `
<div
  className="w-100 dt dt--fixed"
>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={1}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue b"
      id={1}
      onClick={[Function]}
    >
      Currently
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={2}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={2}
      onClick={[Function]}
    >
      Hourly
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={3}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={3}
      onClick={[Function]}
    >
      Daily
    </span>
  </div>
          
</div>
`;

exports[`Button text changes class when clicked 2`] = `
<div
  className="w-100 dt dt--fixed"
>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={1}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={1}
      onClick={[Function]}
    >
      Currently
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={2}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue b"
      id={2}
      onClick={[Function]}
    >
      Hourly
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={3}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={3}
      onClick={[Function]}
    >
      Daily
    </span>
  </div>
          
</div>
`;

exports[`Button text changes class when clicked 3`] = `
<div
  className="w-100 dt dt--fixed"
>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={1}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={1}
      onClick={[Function]}
    >
      Currently
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={2}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue"
      id={2}
      onClick={[Function]}
    >
      Hourly
    </span>
  </div>
  <div
    className="cursor-pointer dtc tc pv2 bg-black-10"
    id={3}
    onClick={[Function]}
  >
    <span
      className="cursor-pointer link underline-hover dark-blue b"
      id={3}
      onClick={[Function]}
    >
      Daily
    </span>
  </div>
          
</div>
`;
